sql/tasks.py

```diff
@@ -15,126 +15,78 @@ class SQLTasks(SQlDriver):
-        session = self.Session()
-        try:
-            query = session.query(Task).filter_by(user_id=user_id).all()
-            return entity_to_dict(query)
-        except:
-            raise
-        finally:
-            session.close()
+        query = self.temp_session.query(Task).filter_by(user_id=user_id).all()
+        return entity_to_dict(query)
 
+    @helper_session
     def get_existing_user_tasks(self, user_id):
-        session = self.Session()
-        try:
-            query = session.query(Task).filter_by(user_id=user_id).filter(
-                Task.status != 'deleted').all()
-            return entity_to_dict(query)
-        except:
-            raise
-        finally:
-            session.close()
+        query = self.temp_session.query(Task).filter_by(user_id=user_id).filter(
+            Task.status != 'deleted').all()
+        return entity_to_dict(query)
 
+    @helper_session
     def get_tasks_for_period(self, user_id, year, month=None, day=None):
-        session = self.Session()
-        try:
-            query = session.query(Task).filter_by(user_id=user_id).filter(
-                        extract('year', Task.calendar_date) == year,
-                        True if month is None else
-                        (extract('month', Task.calendar_date) == month),
-                        True if day is None else
-                        (extract('day', Task.calendar_date) == day)).all()
-            return entity_to_dict(query)
-        except:
-            raise
-        finally:
-            session.close()
+        query = self.temp_session.query(Task).filter_by(user_id=user_id).filter(
+                    extract('year', Task.calendar_date) == year,
+                    True if month is None else
+                    (extract('month', Task.calendar_date) == month),
+                    True if day is None else
+                    (extract('day', Task.calendar_date) == day)).all()
+        return entity_to_dict(query)
 
+    @helper_session
     def get_tasks_counts_for_year(self, user_id, year):
-        session = self.Session()
-        try:
-            query = session.query(
-                func.month(Task.calendar_date), func.count(Task.calendar_date))\
-                .filter_by(user_id=user_id)\
-                .filter(extract('year', Task.calendar_date) == year)\
-                .filter(or_(Task.status == 'active', Task.status == 'done'))\
-                .group_by(func.month(Task.calendar_date)).all()
-            return query
-        except:
-            raise
-        finally:
-            session.close()
+        query = self.temp_session.query(
+            func.month(Task.calendar_date), func.count(Task.calendar_date))\
+            .filter_by(user_id=user_id)\
+            .filter(extract('year', Task.calendar_date) == year)\
+            .filter(or_(Task.status == 'active', Task.status == 'done'))\
+            .group_by(func.month(Task.calendar_date)).all()
+        return query

```

Comment:
Why do not you use _entity to dict(query)_ here?

Reply:
Because this query returns list of tuples, as discussed

Reply:
ok